legate/driver/main.py

```diff
@@ -79,29 +89,5 @@ def canonical_main(argv: list[str]) -> Tuple[Tuple[str, ...], Dict[str, str]]:
             Legion command and env of the driver
 
     """
-    from ..util.system import System
-    from ..util.ui import error
-    from . import CanonicalDriver, Config
-    from .driver import print_verbose
-
-    try:
-        config = Config(argv)
-    except Exception as e:
-        print(error("Could not configure canonical driver:\n"))
-        raise e
-
-    try:
-        system = System()
-    except Exception as e:
-        print(error("Could not determine System settings: \n"))
-        raise e
-
-    try:
-        driver = CanonicalDriver(config, system)
-    except Exception as e:
-        msg = "Could not initialize canonical driver, path config and exception follow:"  # noqa
-        print(error(msg))
-        print_verbose(system)
-        raise e
-
+    driver = prepare_driver(argv, CanonicalDriver)
```

Comment:
I would suggest getting rid of `canonical_main` altogether (it doesn't really do what a "main" function usually does) Instead. Just have the code up in `legate/core/__init__.py` call  `prepare_driver(argv, CanonicalDriver)` itself directly.

Reply:
I see, but isn't it cleaner to wrap `prepare_driver` into `legate_main` and `canonical_main` and then we can call `canonical_main` directly inside `legate/core/__init__.py`?

Reply:
I don't personally think it is cleaner, since calling a function like this "main" seems confusing to me (usually a main function starts a blocking loop event loop, etc and runs for the duration of execution). It also seems superfluous since `canonical_main` really doesn't do anything else at all apart from call `prepare_driver`.

Reply:
OK, do you think if we should remove `legate_main`? The `legate_main` only calls `prepare_driver` as well.

Reply:
I would personally keep it, because it is responsible for calling `driver.run()` on the result of `prepare_driver` (which is a blocking API call to "start" the application, and that is actually want a "main" is typically responsible for)